source/sas.cpp

```diff
@@ -311,8 +314,7 @@ void SAS::Connection::writer()
   }
 }
 
-
-bool SAS::Connection::connect_init()
+int SAS::Connection::get_local_sock()
```

Comment:
This function uses the `SEND_TIMEOUT` constant to set the `SO_SNDTIMEO` socket option. Should a socket factory also set this socket option? How would it get access to the value? Maybe the code in `connect_init` should also set this option to override a potentially wrong value set in the socket_factory?